### PR TITLE
VACMS-14703: VBA facility service changes

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #14703

## Testing done
Manually

#### Node:edit
14703-term-description-VBA-Facility-service-node-edit

#### Node:view
14703-term-description-VBA-Facility-service-node-view

### VAMC System Health Service 
Pre-existing functionality that may have been impacted by necessary refactoring (but was not)
14703-term-description-VAMC-System-Health-service

### Vet Center Facility
Pre-existing functionality that may have been impacted by necessary refactoring (but was not)
14703-term-description-Vet-Center

### Create a VBA Facility service
- [x] Create a [Home Loan](https://pr15356-z1gtfa2un9apjfsmnznfrmtunqashqqc.ci.cms.va.gov/node/add/vba_facility_service) VBA service for Cheyenne VA Regional Benefit office.
- [x] Confirm that the term fields show in node:edit
- [x] Save
- [x] Confirm that the term description shows in node:view

### Create a VAMC System Health Service
- [x] Create an Adaptive sports [VAMC System Health Service](https://pr15356-z1gtfa2un9apjfsmnznfrmtunqashqqc.ci.cms.va.gov/node/add/regional_health_care_service_des) for Black Hills VA health care
- [x] Try a few different terms in the Service name, but land on "Adaptive Sports"
- [x] Confirm that the grey box populates with the appropriate term fields
- [x] Save
- [x] Go back to the node:edit
- [x] Confirm that the grey box populates with the appropriate term fields

### View a Vet Center
- [x] Visit the [Phoenix Vet Center](https://pr15356-z1gtfa2un9apjfsmnznfrmtunqashqqc.ci.cms.va.gov/phoenix-vet-center)
- [x] Confirm that the services are showing with term descriptions
- [x] Click Edit
- [x] Add a new service
- [x] Choose a term description
- [x] Confirm that a national term description shows in the UI

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js [hunk 1]
```diff
@@ -27,7 +27,20 @@
...
+        if (context.getElementById("edit-field-body-wrapper") !== null) {
+          wysiwyg = context.getElementById("edit-field-body-wrapper");
+          nationalEditor = "VHA";
+          // VBA Facility service has "edit-group-service-details"
+        } else if (
+          context.getElementById("edit-group-service-details") !== null
+        ) {
+          wysiwyg = context.getElementById("edit-group-service-details");
+          nationalEditor = "VBA";
+        }
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js:42]: Does this seem brittle?  It looks like we are determining VHA or VBA based on things that are not directly related.
Can we sniff content type?

File: docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js [hunk 2]
```diff
@@ -49,14 +62,13 @@
         const div = context.createElement("div");
         const button = context.createElement("button");
         button.className = "tooltip-toggle css-tooltip-toggle";
-        button.value =
-          "Why can't I edit this? VHA keeps these descriptions standardized to help Veterans identify the services they need.";
+        button.value = `Why can't I edit this? ${nationalEditor} keeps these descriptions standardized to help Veterans identify the services they need.`;
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js:65]: Nice

File: docroot/modules/custom/va_gov_facilities/src/EventSubscriber/FacilitiesSubscriber.php [hunk 3]
```diff
@@ -505,14 +517,51 @@ public function getProductTypeTermFields(array &$form, FormStateInterface $form_
...
+  /**
+   * Gets the VA Services field names for each type of product.
+   *
+   * @param string $node_type
+   *   The type of node.
+   *
+   * @return array
+   *   The field names of the service of the node type.
+   */
+  public function getVaServicesTaxonomyFieldNames(string $node_type) : array {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_facilities/src/EventSubscriber/FacilitiesSubscriber.php:534]: I like the way your broke this out.  Nice work.

File: docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js (hunk 1 above)
Comment (omahane) [reply docroot/modules/custom/va_gov_facilities/js/displayServiceDescriptions.es6.js:42]: It is brittle and is only really necessary for hiding the WYSIWYG on VAMC System Health services and Service Locations field until a user chooses a term, though I don't think that's particularly useful or necessary. I could get rid of it and genericize the "Why can't I edit this?" and thus reduce the brittle nature.
Screenshot 2023-09-25 at 12 01 11 PM

**Comment by swirtSJW** [comment]
I noticed that on VAMCs when you select to use the facility hours, node View actually shows the facility hours
image

I would expect that VBA would pick that up too. 

image

**Comment by omahane** [comment]
The hours display discrepancy is a result of removing the `field_facility_location` from VBA.

Screenshot 2023-09-26 at 8 31 52 AM

**Comment by swirtSJW** [comment]
I just ran into this error creating a VBA service.
image

I can't tell you yet which of my test pieces of data caused that.   I am trying to narrow that down now.

**Comment by swirtSJW** [comment]
I have tried 30 different variations of data to try to reproduce that ^^ but have been unable to.

**Review by swirtSJW — APPROVED** [review]
This tests out great.  Other than the one wierd error that I cannot reproduce this  seems great.

**Comment by swirtSJW** [comment]
image
